Update installation page and contributing guide (#1774)
scverse/scvi-tools: docs/installation.md

## Prerequisites

scvi-tools can be installed via conda or pip. If you don't know which to choose, we recommend conda.

### conda prerequisites

> 1. Install conda. We typically use the [mambaforge] distribution. Use Python version >=3.8, and consider using mamba instead of conda. Mamba is a drop-in replacement for conda that is significantly more efficient.

Reviewer: ```suggestion
1. Install conda. We typically use the [mambaforge] distribution. Use python version >3.7, and consider using mamba instead of conda. Mamba is a drop-in replacement for conda that is significantly more efficient.
```
so that it is consistent with contributing section.

> 1. Install conda. We typically use the [mambaforge] distribution. Use Python version >=3.8, and consider using mamba instead of conda. Mamba is a drop-in replacement for conda that is much faster and more efficient.

Author: ```suggestion
1. Install conda. We typically use the [mambaforge] distribution. Use Python version >=3.8, and consider using mamba instead of conda. Mamba is a drop-in replacement for conda that is significantly more efficient.
```

> 1. Install conda. We typically use the [mambaforge] Python distribution. Use Python version >=3.8, and consider using mamba instead of conda. Mamba is a drop-in replacement for conda that is much faster and more efficient.

Author: ```suggestion
1. Install conda. We typically use the [mambaforge] distribution. Use Python version >=3.8, and consider using mamba instead of conda. Mamba is a drop-in replacement for conda that is much faster and more efficient.
```

### pip prerequisites

0. If using conda/mamba, then just run `conda install -c anaconda pip` and skip this section.
1. Install [Python], we prefer the [pyenv](https://github.com/pyenv/pyenv/) version management system, along with [pyenv-virtualenv](https://github.com/pyenv/pyenv-virtualenv/).
2. Install [PyTorch] and [jax]. If you have an Nvidia GPU, be sure to install versions of PyTorch and jax that support it -- scvi-tools runs much faster with a discrete GPU.

### Apple silicon prerequisites

> Installing scvi-tools on a Mac with Apple Silicon is only possible using a native version of Python. A native version of Python can be installed with an Apple Silicon version of mambaforge (which can be installed from a native version of homebrew via `brew install --cask mambaforge`). This is due to an scvi-tools dependency on jax, which cannot be run via Rosetta.

Reviewer: Should we also note here that PyTorch is not fully supported yet with metal backend?

> Installing scvi-tools on a Mac with Apple Silicon is only possible using a native version of Python. A native version of Python can be installed with an Apple Silicon version of mambaforge (which can be installed from a native version of homebrew via `brew install --cask mambaforge`). This is due to an scvi-tools dependency on jax, which cannot be run via Rosetta.

Reviewer: ```suggestion
Installing scvi-tools on a Mac with Apple Silicon is only possible using a native version of python. A native version of python can be installed with an Apple Silicon version of mambaforge (which can be installed from a native version of homebrew via `brew install --cask mambaforge`). This is due to an scvi-tools dependency on jax, which cannot be run via Rosetta.
```

### Windows prerequisites

> After setting up a virtual environment with conda/mamba/pyenv, please install the [community-buit version of jax](https://github.com/cloudhan/jax-windows-builder) before installing scvi-tools.

Reviewer: ```suggestion
After setting up a virtual environment with conda/mamba/pyenv, please install the [community-built version of jax](https://github.com/cloudhan/jax-windows-builder) before installing scvi-tools.
```

```
pip install "jax[cpu]" -f https://whls.blob.core.windows.net/unstable/index.html --use-deprecated legacy-resolver
```

### GPU prerequisites

> All scvi-tools models will be faster when accelerated with a GPU. Before installing scvi-tools, you can install GPU versions of PyTorch and Jax using conda as follows:

Reviewer: ```suggestion
All scvi-tools models will be faster when accelerated with a GPU. Before installing scvi-tools, you can install GPU versions of PyTorch and jax using conda as follows:
```

### pip prerequisites

1. Install [Python], we prefer the [pyenv](https://github.com/pyenv/pyenv/) version management system, along with [pyenv-virtualenv](https://github.com/pyenv/pyenv-virtualenv/).

> 2. Install [PyTorch] and [jax]. If you have an Nvidia GPU, be sure to install a versions of PyTorch and jax that support it -- scvi-tools runs much faster with a discrete GPU.

Author: ```suggestion
2. Install [PyTorch] and [jax]. If you have an Nvidia GPU, be sure to install versions of PyTorch and jax that support it -- scvi-tools runs much faster with a discrete GPU.
```
